frontend/src/metabase/query_builder/components/template_tags/CardTagEditor.jsx

```diff
@@ -0,0 +1,150 @@
+    setDatasetQuery(
+      query.replaceCardId(question ? question.id : "", id).datasetQuery(),
+    );
+    this._popover && this._popover.close();
+  };
+
+  errorMessage() {
+    const { error, question, query } = this.props;
+
+    if (
+      question &&
+      // If this question was loaded by a search endpoint before fetching, it
+      // might not have a database_id set yet.
+      question.database_id != null &&
+      question.database_id !== query.databaseId()
+    ) {
+      return t`This question can't be used because it's based on a different database.`;
+    }
+    if (error) {
+      return error.status === 404
+        ? t`Couldn't find a saved question with that ID number.`
+        : error.data;
+    }
+    return null;
+  }
+
+  triggerElement() {
+    const { id, question } = this.props;
+    return (
+      <SelectButton>
+        {id == null ? (
+          <span className="text-medium">{t`Pick a saved question`}</span>
+        ) : this.errorMessage() ? (
+          <span className="text-medium">{t`Pick a different question`}</span>
+        ) : question ? (
+          question.name
+        ) : (
+          // we only hit this on the initial render before we fetch
+          t`Loading…`
+        )}
+      </SelectButton>
+    );
+  }
+
+  render() {
+    const { id, loading, question } = this.props;
+
+    return (
+      <Card className="p2 mb2">
+        <h3 className="text-brand mb2">
+          {id == null ? (
+            t`Question #…`
+          ) : (
+            <Link to={questionUrl(id)}>{t`Question #${id}`}</Link>
+          )}
+        </h3>
+        {loading ? (
+          <LoadingSpinner />
+        ) : (
+          <PopoverWithTrigger
+            ref={ref => (this._popover = ref)}
+            triggerElement={this.triggerElement()}
+            verticalAttachments={["top", "bottom"]}
+            pinInitialAttachment
+          >
+            <QuestionPicker
+              className="p2"
+              value={question && question.id}
+              onChange={this.handleQuestionSelection}
+            />
+          </PopoverWithTrigger>
+        )}
+        {this.errorMessage() && (
+          <p className="text-error bg-light p2 mt2 mb0">
+            {this.errorMessage()}
+          </p>
+        )}
+        {question && !this.errorMessage() && (
+          <div className="bg-light text-medium p1 mt2">
+            {question.collection && (
+              <div>
+                <Icon name="all" size={12} className="mr1" />{" "}
+                {question.collection.name}
+              </div>
+            )}
+            <div>
+              <Icon name="calendar" size={12} className="mr1" />{" "}
+              {t`Last edited ${formatDate(question.updated_at)}`}
+            </div>
+          </div>
+        )}
+      </Card>
+    );
+  }
+}
+
+// This formats a timestamp as a date using any custom formatting options.
+function formatDate(value) {
```

Comment:
This should probably be in `metabase/lib/formatting` but currently that code always assumes the custom formatting is passed in via `column` settings, so maybe this is ok for now.

Reply:
Yeah, it is formatting code, but it's not for question data. 🤷‍♂Are there any other places in the app where we want to use general formatting settings for "UI data"?

Also, these timestamps seem to come down without TZ data, so there's probably some issues there.

Reply:
There are some places we use `formatNumber` (maybe others) on non-results, but yeah, I'm not sure we should be conflating the two use cases.